File: backend/src/main/java/ru/haliksar/tictactoe/backend/model/RoomPlayer.java
package ru.haliksar.tictactoe.backend.model;

import javax.persistence.Entity;
import javax.persistence.Id;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@Entity(name = "RoomPlayer")
@EqualsAndHashCode(of = {"id"})
public class RoomPlayer {
    @Id
    private String id;

    private Marker marker;

    private boolean win;

    private boolean move;

    private String nickname;
}

File: app/src/main/java/ru/haliksar/tictactoe/domain/entity/RoomPlayer.kt
package ru.haliksar.tictactoe.domain.entity

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class RoomPlayer(
    @Json(name = "id") val id: String,
    @Json(name = "marker") val marker: Marker
)
